features/FxA-107-security-event-history/README.md

```diff
@@ -0,0 +1,74 @@
+# Security Events History
+
+## Problem Summary
+
+As a Firefox Accounts user, I want to reduce the amount of sign-in
+confirmations I must make, especially when I am logging in from the same
+locations each time (my office and/or home).
+
+--------
+
+## Outcomes
+
+To start, report whenever a sign in attempt that would trigger signin
+confirmation could have been elided, based on correlating the current
+signin request with the user's history.
+
+## Metrics
+
+We can log an info message each time this would occur, tagged as
+`Account.history.seen`. A graph can be created, comparing:
+
+- Number of signin requests
+- Number of `Account.history.seen`
+
+--------
+
+## Detailed design
+
+> This is the bulk of the RFC. Explain the design in enough detail for somebody familiar
+with the language to understand.
+This should get into specifics and corner-cases, and include examples of how the feature is used.
+
+Keep a history of "security" events that occur for a user. A record in
+this history would contain these details:
+
+- `uid` - user ID from FxA
+- An event name (`account.login`, `account.reset`, `account.confirm`, etc)
+- IP Address of user
+- Timestamp
```

Comment:
It all depends how quickly you want to release v1 of this feature. Geo is hard and calculating the safe connection area of a user means dealing with tons of edge cases and inaccuracies from geo databases. I would recommend to leave it out of v1 and only compare hmacs of IPs for now. That means users who moved to the nearby coffee shop will be seen as untrusted, but it's probably ok for now?

Then for v2 we can design a geo algorithm that doesn't suck too much. Still, the rate of false positives will be in the double digits based on past experiences implementing this.

If you decide to go full geo in v1, let me know and I'll share prior work.